akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -6,10 +6,28 @@ package akka.stream
 import scala.collection.immutable
 import scala.collection.JavaConverters._
 
+/**
+ * An input port of a StreamLayout.Module. This type logically belongs
+ * into the impl package but must live here due to how `sealed` works.
```

Comment:
The comment is a bit confusing TBH - maybe explain why exactly, or make it `//` comments?